cellxgene_schema_cli/tests/test_schema_compliance.py

```diff
@@ -0,0 +1,1006 @@
+        # this MUST be either a HANCESTRO term or "unknown" if unavailable.
+        self.validator.adata.obs["organism_ontology_term_id"][0] = "NCBITaxon:9606"
+        self.validator.adata.obs["ethnicity_ontology_term_id"][0] = "EFO:0000001"
+        self.validator.validate_adata()
+        self.assertEqual(
+            self.validator.errors,
+            [
+                "ERROR: 'EFO:0000001' in 'ethnicity_ontology_term_id' is "
+                "not a valid ontology term id of 'HANCESTRO'"
+            ],
+        )
+
+        # Otherwise, for all other organisms this MUST be "na".
+        self.validator.errors = []
+        self.validator.adata.obs["organism_ontology_term_id"][0] = "NCBITaxon:10090"
+        self.validator.adata.obs["development_stage_ontology_term_id"][
+            0
+        ] = "MmusDv:0000003"
+        self.validator.adata.obs["ethnicity_ontology_term_id"][0] = "EFO:0000001"
+        self.validator.validate_adata()
+        self.assertEqual(
+            self.validator.errors,
+            [
+                "ERROR: 'EFO:0000001' in 'ethnicity_ontology_term_id' is not a "
+                "valid value of 'ethnicity_ontology_term_id'"
+            ],
+        )
+
+    def test_organism_ontology_term_id(self):
+
+        """
+        organism_ontology_term_id categorical with str categories. This MUST be a child of NCBITaxon:33208.
+        """
+
+        # If organism_ontolology_term_id is "NCBITaxon:9606" for Homo sapiens,
+        # this MUST be either a HANCESTRO term or "unknown" if unavailable.
+        self.validator.adata.obs["organism_ontology_term_id"][0] = "EFO:0000001"
+        self.validator.adata.obs["development_stage_ontology_term_id"][0] = "unknown"
+        self.validator.adata.obs["ethnicity_ontology_term_id"][0] = "na"
+        self.validator.validate_adata()
+        self.assertEqual(
+            self.validator.errors,
+            [
+                "ERROR: 'EFO:0000001' in 'organism_ontology_term_id' is not a valid "
+                "ontology term id of 'NCBITaxon'"
+            ],
+        )
+
+    def test_tissue_ontology_term_id(self):
+
+        """
+        tissue_ontology_term_id categorical with str categories. This MUST be the term that best describes the tissue
+        that this cell was derived from, depending on the type of biological sample:
+            Cell Culture - MUST be a CL term appended with " (cell culture)"
+            Organoid - MUST be an UBERON term appended with " (organoid)"
+            Enriched, Sorted,or Isolated Cells from a Tissue - MUST be an UBERON or CL term and SHOULD NOT use terms
+                that do not capture the tissue of origin (e.g. In the case of CD3+ kidney cells, use "UBERON:0002113"
+                for kidney instead of "CL:000084" for T cell. However, in the case of EPCAM+ cervical cells,
+                use "CL:000066" for epithelial cell of the cervix.)
+        """
```

Comment:
There's a lot of constraints here, but it seems they cannot be programmatically verified here, since we don't know the type of the biological sample, right?

Reply:
[`tissue_ontology_term_id`](https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/2.0.0/corpora_schema.md) started with a simple MUST for UBERON and then evolved over time for cases that cannot be programmatically verified. 

The final _Enriched, Sorted, or Isolated Cells from a Tissue_ case is the proverbial straw. It has become more like the STRONGLY RECOMMENDED that we use for `development_stage_ontology_term_id`. 

We should probably rewrite to clarify:

1  curie MUST BE UBERON or CL - with STRONGLY RECOMMENDED details for the cases but only enforcing the ontology selection
2. MUST BE UBERON (organoid) - if the curator annotates for organoids
3. MUST BE CL (cell culture) - if the curator annotates for cell cultures

@ambrosejcarr ^^^

Reply:
Correct

Reply:
Addressing in #131

Reply:
Finished in #131